### ISSUE TITLE
Unable to import module from typescript

### ISSUE THREAD

**Issue by mleguen** [jonschlinkert/time-stamp#14]
This is a follow-up of #12 

The proposed solution does not work either:
```
import * as timestamp from 'time-stamp';
```
leads to :
```
error TS2349: This expression is not callable.
  Type 'typeof import("[...]/node_modules/time-stamp/index")' has no call signatures.
```

**Comment by mleguen** [comment]
The only way I found to avoid this error is to add the following line in index.js:
```js
timestamp.default = timestamp;
```
and to switch back in my code to:
```js
import timestamp from 'time-stamp';
```
However I can find no way to write an index.d.ts compatible with the current index.js.

**Comment by jonschlinkert** [comment]
want to do a pr?

**Comment by mleguen** [comment]
@jonschlinkert I do not see at the moment how to do it without impacting too much the current behavior of index.js.

**Comment by fregante** [comment]
If you use `module.exports = foo` in your JS file, the TypeScript definition should be `export = foo`. Example https://github.com/sindresorhus/mem/pull/32

`export default` is only for ES Modules and `time-stamp` isn't one; adding a `default` export won't change it into one.

TypeScript offers `esModuleInterop` but it only lets users of a CJS module to import it via `import mod from 'module'`, this module doesn't need any special syntax to make it compatible.

**Comment by mleguen** [comment]
@fregante OK, thank you. I will experiment on this in the coming days to see if I can do a PR.

**Comment by mleguen** [comment]
PR submitted

**Comment by luckydrq** [comment]
I commented at the PR(https://github.com/jonschlinkert/time-stamp/pull/18#issuecomment-732671198) and make a new PR(https://github.com/jonschlinkert/time-stamp/pull/20) to solve this. In a word, `import foo = require('foo')` need module `foo` have `exports.default` exported, or it would not work.

**Comment by papb** [comment]
> In a word, `import foo = require('foo')` need module `foo` have `exports.default` exported, or it would not work.

This is not true. For `import foo = require('foo')` to work, the module needs to have `export = foo` in the entrypoint.
